### ISSUE TITLE
DistributedConv2d (feature) wrong output shape for 1 worker

### ISSUE THREAD

**Issue by jmerizia** [distdl/distdl#208]
```
# mpirun -n 1 python -m mpi4py <filename>.py

import torch
from mpi4py import MPI
import distdl
import numpy as np


worker_grid_shape = [1, 1]

workers = np.arange(worker_grid_shape[0]*worker_grid_shape[1])
shape = [1, 1, worker_grid_shape[0], worker_grid_shape[1]]

partition_world = distdl.backends.mpi.Partition(MPI.COMM_WORLD)
partition_base = partition_world.create_partition_inclusive(workers)
partition = partition_base.create_cartesian_topology_partition(shape)

conv = distdl.nn.DistributedConv2d(partition, in_channels=3, out_channels=5, kernel_size=(3, 3), padding=(1, 1))

x = torch.rand((1, 3, 64, 64))
y = conv(x)

print(x.shape)
print(y.shape)
```

The above code will produce

```
torch.Size([1, 3, 64, 64])
torch.Size([1, 5, 62, 62])
```

which is incorrect (they should be equal in dimensions 2 and 3).

**Comment by jmerizia** [comment]
Alright, I narrowed down the issue. We handle the serial case manually, and we recently made the choice to handle padding manually. Thus, we must initialize the conv layer in the serial case separately.

https://github.com/distdl/distdl/blob/a07429343f047b6a3352b03c1947cc0df37891d0/src/distdl/nn/conv_feature.py#L126
